webpack/assets/javascripts/react_app/components/common/table/index.js

```diff
@@ -0,0 +1,40 @@
+import { Table as PfTable } from 'patternfly-react';
+import React from 'react';
+import EllipsisWithTooltip from 'react-ellipsis-with-tooltip';
+import EmptyState from '../emptyState';
+
+export const headerFormat = value => <PfTable.Heading>{value}</PfTable.Heading>;
+export const cellFormat = value => <PfTable.Cell>{value}</PfTable.Cell>;
+export const ellipsisFormat = value => (
+  <PfTable.Cell>
+    <EllipsisWithTooltip>{value}</EllipsisWithTooltip>
+  </PfTable.Cell>
+);
+
+class Table extends React.Component {
+  isEmpty() {
+    return this.props.rows.length === 0;
+  }
+
+  render() {
+    const { columns, rows, emptyState } = this.props;
+    return this.isEmpty() ? (
+      <EmptyState {...emptyState} />
+    ) : (
+      <PfTable.PfProvider
+        className="table-fixed"
+        striped
+        bordered
+        hover
+        columns={columns}
+      >
+        <PfTable.Header />
+        <PfTable.Body
+          rows={rows}
+          rowKey={({ rowData, rowIndex }) => rowIndex}
+        />
+      </PfTable.PfProvider>
```

Comment:
Do we want to include pagination with this?  Right now if you have pagination on your table and trigger the empty state you get this:

<img width="1217" alt="screen shot 2018-03-27 at 11 38 50 am" src="https://user-images.githubusercontent.com/4116405/37978179-73707474-31b3-11e8-8ed6-a92f3c44de64.png">

Was thinking maybe you could pass the pagination component into this directly so that way you maintain control of it from within your component.  In other words:

```javascript
const paginationRow = (
  <PaginationRow
    viewType="table"
    itemCount={subscriptions.pagination.total}
    pagination={subscriptions.pagination}
    onChange={onPaginationChange}
/>
);

<Table
  rows={subscriptions.results}
  columns={columns}
  emptyState={emptyStateData()}
  paginationRow={paginationRow}
/>
```

Then all this component would do is include the pagination within the ternary operator.

Reply:
I'm also okay with adding this in a future PR.

Reply:
I'm leaning towards fixing it in a future PR. It's not a blocker for this one that has been open for a long time already.

I'd propose a bit different solution. The empty state design is quite constant while the table implementation can vary a lot (inline edits, provider setup, etc.). I think that it would be more flexible to refactor (and maybe rename) the component so that it accepts children with default being a table. But as I've said, I prefer to solve it outside of this PR.

Reply:
> I'd propose a bit different solution. The empty state design is quite constant while the table implementation can vary a lot (inline edits, provider setup, etc.). I think that it would be more flexible to refactor (and maybe rename) the component so that it accepts children with default being a table. But as I've said, I prefer to solve it outside of this PR.

I also prefer this solution and am okay updating katello once we have it done.